test/unit/attributes.js

```diff
@@ -1633,3 +1633,20 @@ QUnit.test( "SVG class manipulation (gh-2199)", function( assert ) {
 		assert.ok( !elem.hasClass( "awesome" ), "SVG element (" + this + ") toggles the class off" );
 	} );
 } );
+
+QUnit.test( "non-lowercase boolean attribute getters should not crash", function( assert ) {
+	assert.expect( 3 );
+
+	jQuery.each( [
+		"Checked",
+		"requiRed",
+		"SELECTED"
+	], function() {
+	    try {
+			jQuery( "<div>" ).attr( this );
```

Comment:
You're right, I'll change that.

What do you think about the `try-catch` approach? Or should I just check assertions and rely on QUnit to fail when it throws? I thought the `try-catch` approach is better here where I'm specifically testing that no error is thrown.